### PR TITLE
feat: add ASH, Bandit and CDK Nag

### PR DESCRIPTION
**Pull request by jaidisido**
### Feature or Bugfix
- Feature

### Detail
Add ASH, Bandit and CDK Nag security checks.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jaidisido — COMMENTED** [review]
Some initial comments, still WIP

File: .github/workflows/cdk-nag.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: CDK Nag
```
[the remark this answers is not in the document]
Comment (jaidisido) [reply .github/workflows/cdk-nag.yml:1]: I confirm this is possible via an [OIDC](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services) connection

**Comment by jaidisido** [comment]
> The new Bandit job fails due to 2 occurences of Popen calls with the argument "shell=True" which basically makes the process to be executed via shell. We have a couple of other occurences in the repo where this finding is suppressed with the "# nosec" tag, eg. in the [cdk_cli_wrapper](https://github.com/awslabs/aws-dataall/blob/main/backend/dataall/cdkproxy/cdk_cli_wrapper.py#L161)
> 
> The [bandit guideline](https://bandit.readthedocs.io/en/1.7.5/plugins/b602_subprocess_popen_with_shell_equals_true.html) suggests that the input which is executed in the shell should be properly sanitized / checked to mitigate the risk.
> 
> @dlpzx How do you want to proceed? Should we simply suppress this warning to get this PR merged, or do you want to have first some input sanitization steps to be implemented, everywhere where this "shell=True" option is used?

@dbalintx thanks for the input. Agreed - the # nosec comments should be removed and some time should be spent by the team implementing input sanitization for all subprocess calls. [semgrep](https://semgrep.dev/docs/cheat-sheets/python-command-injection/) has a guide with mitigation suggestions for example

**Comment by dlpzx** [comment]
I agree @dbalintx , we should fix and not ignore the issues. In this other PR #536 I added semgrep and it points out to the same issue

**Comment by dlpzx** [comment]
@noah-paige @jaidisido, I added some #nosec comments for the shell-true findings. There is a separate [issue](https://github.com/awslabs/aws-dataall/issues/738) to track the fix, which has been added to milestone 2.1.0

**Pull request merged**